# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/gfx-backend-empty/src/image.rs
# Commit: ef5dc3e04e5f
# Full Hash: ef5dc3e04e5f271eea0636ab3a495e95cc912f1d
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2021-09-04 09:40:24
# Regressor Bug: 1726626
# File Overlap Count: 3
# Description:
#   Bug 1726626 - Move gfx/wgpu into a 3rd party dependency r=jgilbert,bholley
#   
#   This update makes wgpu a vendored dependency instead of having it in gfx/wgpu.
#   
#   ## Notes
# ==============================================================================

diff --git a/third_party/rust/gfx-backend-empty/src/image.rs b/third_party/rust/gfx-backend-empty/src/image.rs
deleted file mode 100644
--- a/third_party/rust/gfx-backend-empty/src/image.rs
+++ /dev/null
@@ -1,30 +0,0 @@
-use hal::image::Kind;
-use hal::memory::Requirements as MemoryRequirements;
-
-#[derive(Debug)]
-pub struct Image {
-    /// What type of image this is, as well as its extent.
-    kind: Kind,
-}
-
-impl Image {
-    pub fn new(kind: Kind) -> Self {
-        Image { kind }
-    }
-
-    pub fn get_requirements(&self) -> MemoryRequirements {
-        let size = match self.kind {
-            Kind::D2(width, height, layers, samples) => {
-                assert_eq!(layers, 1, "Multi-layer images are not supported");
-                assert_eq!(samples, 1, "Multisampled images are not supported");
-                u64::from(width) * u64::from(height)
-            }
-            _ => unimplemented!("Unsupported image kind"),
-        };
-        MemoryRequirements {
-            size,
-            alignment: 1,
-            type_mask: !0,
-        }
-    }
-}